src/components/PopupWithForm.js

```diff
@@ -5,6 +5,9 @@ export default class PopupWithForm extends Popup {
     super(popupSelector);
     this._popupForm = this.popupElement.querySelector(".popup__form");
     this._handleFormSubmit = handleFormSubmit;
+    // Store the initial text content of the submit button
+    this._submitButtonText =
+      this._popupForm.querySelector(".popup__button").textContent;
   }
   _getInputValues() {
     const inputList = Array.from(
```

Comment:
**Could be improved**

All elements that are found every time you call a method should be defined as **class fields** in the constructor, `_setEventListeners` (or in `generateCard`) only **1 time** not to waste resources on searching them again and again.

`Class fields` (the so called **class variables** with `this`) are accessible inside any method of the class. You shouldn’t pass **class fields** into the call of any method or search them again because you can use class fields inside all methods.
There can be `this._inputs`, `this._submitButton`, `this._cardImage`, `this._form` , `this._likeButton` and so on.